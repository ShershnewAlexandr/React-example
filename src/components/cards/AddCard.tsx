import React from 'react';
import { Link } from 'react-router-dom';
import { Grid } from '@material-ui/core';
import { routes } from '../../utils/constants';
import './CardsStyles.css';

function AddCard() {
  return (
    <div className="add-card-container">
      <Grid
        className="card-space"
        container
        justify="center"
        alignItems="center"
      >
        <Link to={routes.NEWCARD} className="link">
          <Grid item>
            <div className="card-plus-container">
              <Grid container justify="center" alignItems="center">
                <Grid item>
                  <span className="card-plus">+</span>
                </Grid>
              </Grid>
            </div>
          </Grid>
        </Link>
      </Grid>
    </div>
  );
}

export default AddCard;
