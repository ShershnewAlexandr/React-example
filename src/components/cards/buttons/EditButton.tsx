import React from 'react';
import { Link } from 'react-router-dom';
import { Grid } from '@material-ui/core';
import { routes } from '../../../utils/constants';
import '../CardsStyles.css';

interface EditButtonProps {
  id: number;
  setEditPostAction: any;
}

function EditButton(props: EditButtonProps) {
  const { id, setEditPostAction } = props;
  return (
    <Link
      to={routes.EDITCARDID(id)}
      onClick={() => {
        setEditPostAction(id);
      }}
    >
      <Grid
        container
        className="edit-button-container"
        justify="center"
        alignItems="center"
      >
        <Grid item>
          <i className="material-icons md-24">edit</i>
        </Grid>
      </Grid>
    </Link>
  );
}

export default EditButton;
