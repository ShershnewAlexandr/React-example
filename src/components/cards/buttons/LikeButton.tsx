import React from 'react';
import { Grid } from '@material-ui/core';
import '../CardsStyles.css';

interface LikeButtonProps {
  id: number;
  liked: boolean;
  totalLikes: number;
  likePostAction: any;
}

interface LikeButtonState {
  liked: boolean;
}

class LikeButton extends React.Component<LikeButtonProps, LikeButtonState> {
  constructor(props) {
    super(props);
    this.state = {
      liked: props.liked,
    };
  }

  onLike = () => {
    const { id, likePostAction } = this.props;
    this.setState({
      liked: !this.state.liked,
    });
    likePostAction(id);
  };

  render() {
    const { totalLikes } = this.props;
    const { liked } = this.state;
    return (
      <Grid
        container
        direction="row"
        className="like-button-container"
        justify="space-around"
        alignItems="center"
        data-testid="likeButton"
        onClick={this.onLike}
      >
        <Grid item>
          <span>{totalLikes}</span>
        </Grid>
        <Grid item>
          <i
            data-testid="like-icon"
            className={
              liked ? 'material-icons md-24 liked' : 'material-icons md-24'
            }
          >
            thumb_up
          </i>
        </Grid>
      </Grid>
    );
  }
}

export default LikeButton;
