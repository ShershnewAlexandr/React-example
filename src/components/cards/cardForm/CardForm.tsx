import React from 'react';
import { Grid } from '@material-ui/core';
import { Field } from 'redux-form/immutable';
import { InjectedFormProps } from 'redux-form';
import TextInput from '../../inputs/TextInput';
import TextAreaInput from '../../inputs/TextAreaInput';
import { useTranslation } from 'react-i18next';

function CardForm(props: InjectedFormProps) {
  const { t } = useTranslation();
  const { handleSubmit } = props;

  return (
    <form onSubmit={handleSubmit}>
      <Grid container direction="row" spacing={4}>
        <Grid item xs={12}>
          <Field
            name="title"
            component={TextInput}
            label={t('cardForm.title')}
            disabled={false}
          />
        </Grid>
        <Grid item xs={12}>
          <Field
            name="content"
            component={TextAreaInput}
            type="date"
            label={t('cardForm.content')}
            disabled={false}
            rows={6}
          />
        </Grid>
      </Grid>
    </form>
  );
}

export default CardForm;
