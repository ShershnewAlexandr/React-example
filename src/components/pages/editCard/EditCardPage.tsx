import React from 'react';
import { Grid } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import EditCardFormContainer from '../../../containers/editCard/EditCardFormContainer';
import { EditCardPageContainerProps } from '../../../containers/editCard/EditCardPageContainer';
import CardLoader from '../../loaders/CardLoader';
import { routes } from '../../../utils/constants';
import './EditCardPageStyles.css';

interface EditCardPageOwnProps {
  id: number;
}

function EditCardPage(
  props: EditCardPageContainerProps & EditCardPageOwnProps,
) {
  const { t } = useTranslation();
  const { editPostAction, deletePostAction, id, isLoading } = props;

  return (
    <Grid container alignItems="center" justify="center">
      <Grid
        item
        className="editCardWrapper-container"
        container
        direction="column"
        justify="space-around"
        alignItems="center"
        spacing={2}
      >
        <Grid item>
          <span className="editCardWrapper-title">
            {t('editCard.cardInfo')}
          </span>
        </Grid>
        <Grid item>
          <EditCardFormContainer />
        </Grid>
        <Grid
          container
          item
          xs={12}
          alignItems="center"
          justify="center"
          spacing={2}
        >
          {isLoading ? (
            <CardLoader />
          ) : (
            <>
              <Grid item>
                <button
                  className="editCardWrapper-saveButton"
                  type="submit"
                  onClick={() => editPostAction(id)}
                >
                  {t('editCard.save')}
                </button>
              </Grid>
              <Grid item>
                <button
                  className="editCardWrapper-deleteButton"
                  type="submit"
                  onClick={() => deletePostAction(id)}
                >
                  {t('editCard.delete')}
                </button>
              </Grid>
              <Grid item>
                <Link to={routes.CARDLIST}>
                  <button
                    className="editCardWrapper-CancelButton"
                    type="submit"
                  >
                    {t('editCard.cancel')}
                  </button>
                </Link>
              </Grid>
            </>
          )}
        </Grid>
      </Grid>
    </Grid>
  );
}

export default EditCardPage;
