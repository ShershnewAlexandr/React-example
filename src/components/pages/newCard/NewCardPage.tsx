import React from 'react';
import { Grid } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import NewCardFormContainer from '../../../containers/newCard/NewCardFormContainer';
import { NewCardPageContainerProps } from '../../../containers/newCard/NewCardPageContainer';
import CardLoader from '../../loaders/CardLoader';
import { routes } from '../../../utils/constants';
import './NewCardPageStyles.css';

function NewCardPage(props: NewCardPageContainerProps) {
  const { t } = useTranslation();
  const { addPostAction, isLoading } = props;

  return (
    <Grid container alignItems="center" justify="center">
      <Grid
        item
        className="newCardWrapper-container"
        container
        direction="column"
        justify="space-around"
        alignItems="center"
        spacing={2}
      >
        <Grid item>
          <span className="newCardWrapper-title">{t('newCard.newCard')}</span>
        </Grid>
        <Grid item>
          <NewCardFormContainer />
        </Grid>
        <Grid
          container
          item
          xs={12}
          alignItems="center"
          justify="center"
          spacing={2}
        >
          {isLoading ? (
            <CardLoader />
          ) : (
            <>
              <Grid item>
                <button
                  onClick={addPostAction}
                  className="newCardWrapper-addButton"
                  type="submit"
                >
                  {t('newCard.add')}
                </button>
              </Grid>
              <Grid item>
                <Link to={routes.CARDLIST}>
                  <button
                    className="editCardWrapper-CancelButton"
                    type="submit"
                  >
                    {t('newCard.cancel')}
                  </button>
                </Link>
              </Grid>
            </>
          )}
        </Grid>
      </Grid>
    </Grid>
  );
}

export default NewCardPage;
