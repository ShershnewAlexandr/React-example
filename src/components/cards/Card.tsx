import React from 'react';
import { Grid } from '@material-ui/core';
import EditButton from './buttons/EditButton';
import LikeButton from './buttons/LikeButton';
import { ICard } from '../../types/commonTypes';
import './CardsStyles.css';

interface CardOwnProps {
  isAdmin: boolean;
  likePostAction: any;
  setEditPostAction: any;
}

function Card(props: ICard & CardOwnProps) {
  const {
    id,
    title,
    content,
    total_likes,
    liked,
    likePostAction,
    setEditPostAction,
    isAdmin,
  } = props;
  return (
    <div className="card-container">
      <Grid
        container
        direction="column"
        justify="space-between"
        alignItems="flex-start"
        spacing={1}
      >
        <Grid item className="card-title-container">
          <span className="card-title-text">{title}</span>
        </Grid>
        <Grid item className="card-content-container">
          <span className="card-content-text">{content}</span>
        </Grid>
        <Grid
          item
          container
          className="card-buttons-container"
          direction="row"
          justify={isAdmin ? 'space-between' : 'flex-end'}
          alignItems="center"
        >
          {isAdmin && (
            <Grid item data-testid="edit-card-button">
              <EditButton id={id} setEditPostAction={setEditPostAction} />
            </Grid>
          )}
          <Grid item data-testid="like-card-button">
            <LikeButton
              id={id}
              liked={liked}
              totalLikes={total_likes}
              likePostAction={likePostAction}
            />
          </Grid>
        </Grid>
      </Grid>
    </div>
  );
}

export default Card;
