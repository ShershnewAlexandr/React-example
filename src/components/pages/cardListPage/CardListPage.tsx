import React from 'react';
import { Grid } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import AddCard from '../../cards/AddCard';
import Card from '../../cards/Card';
import { ICard } from '../../../types/commonTypes';
import { routes } from '../../../utils/constants';
import { CardListPageContainerProps } from '../../../containers/CardListPageContainer';
import CardPageBarContainer from '../../../containers/CardPageBarContainer';
import './CardListPageStyles.css';

function CardListPage(props: CardListPageContainerProps) {
  const { t } = useTranslation();

  const { isAdmin, likePostAction, setEditPostAction } = props;
  const cards = props.cards.map((info: ICard) => (
    <Grid item key={info.id}>
      <Card
        {...info}
        isAdmin={isAdmin}
        likePostAction={likePostAction}
        setEditPostAction={setEditPostAction}
      />
    </Grid>
  ));

  return (
    <Grid container>
      <Grid
        item
        container
        direction="row"
        alignItems="flex-end"
        justify="flex-end"
      >
        <Grid item sm={1}>
          <Link to={routes.SIGNIN}>
            <span>{t('cardList.signOut')}</span>
          </Link>
        </Grid>
      </Grid>
      <Grid
        item
        className="cardlist-container"
        container
        direction="row"
        justify="space-evenly"
        alignItems="flex-start"
      >
        {isAdmin && (
          <Grid item>
            <AddCard />
          </Grid>
        )}
        {cards}
      </Grid>
      <Grid container justify="center">
        <Grid item>
          <CardPageBarContainer />
        </Grid>
      </Grid>
    </Grid>
  );
}

export default CardListPage;
