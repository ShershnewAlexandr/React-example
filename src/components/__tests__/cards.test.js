import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import CardForm from '../cards/cardForm/CardForm';
import Card from '../cards/Card';
import AddCard from '../cards/AddCard';
import EditButton from '../cards/buttons/EditButton';
import LikeButton from '../cards/buttons/LikeButton';
import CardListPage from '../pages/cardListPage/CardListPage';
import EditCardPage from '../pages/editCard/EditCardPage';
import NewCardPage from '../pages/newCard/NewCardPage';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('cardForm', () => {
  const props = {
    handleSubmit: jest.fn(),
  };

  const cardForm = shallow(<CardForm {...props} />);

  it('snapshot', () => {
    expect(toJson(CardForm)).toMatchSnapshot();
  });

  it('should catch submit', () => {
    cardForm.find('form').simulate('submit');
    expect(props.handleSubmit.mock.calls.length).toBe(1);
  });
});

describe('card', () => {
  const props = {
    id: 5,
    title: 'title',
    content: 'content text',
    total_likes: 89,
    liked: false,
    likePostAction: jest.fn(),
    isAdmin: false,
  };

  const card = shallow(<Card {...props} />);
  const cardAdmin = shallow(<Card {...props} isAdmin={true} />);
  const cardLiked = shallow(<Card {...props} liked={true} />);

  it('snapshots', () => {
    expect(toJson(card)).toMatchSnapshot();
    expect(toJson(cardAdmin)).toMatchSnapshot();
    expect(toJson(cardLiked)).toMatchSnapshot();
  });
});

describe('addCard', () => {
  const addCard = shallow(<AddCard />);

  it('snapshot', () => {
    expect(toJson(addCard)).toMatchSnapshot();
  });
});

describe('editButton', () => {
  const editButton = shallow(<EditButton id={5} />);

  it('snapshot', () => {
    expect(toJson(editButton)).toMatchSnapshot();
  });
});

describe('likeButton', () => {
  const props = {
    id: 45,
    liked: false,
    totalLikes: 23,
    likePostAction: jest.fn(),
  };

  const likeButton = shallow(<LikeButton {...props} />);

  it('should shitch color after like', () => {
    likeButton.find('[data-testid="likeButton"]').simulate('click');
    expect(likeButton.state().liked).toBeTruthy();
    likeButton.find('[data-testid="likeButton"]').simulate('click');
    expect(likeButton.state().liked).toBeFalsy();
  });

  it('snapshot', () => {
    expect(toJson(likeButton)).toMatchSnapshot();
  });
});

describe('cardListPage', () => {
  const props = {
    cards: [],
    isAdmin: false,
  };

  const cardListPage = shallow(<CardListPage {...props} />);
  const cardListPageAdmin = shallow(<CardListPage {...props} isAdmin={true} />);

  it('snapshots', () => {
    expect(toJson(cardListPage)).toMatchSnapshot();
    expect(toJson(cardListPageAdmin)).toMatchSnapshot();
  });
});

describe('editCardPage', () => {
  const editCardPage = shallow(<EditCardPage id={5} />);

  it('snapshot', () => {
    expect(toJson(editCardPage)).toMatchSnapshot();
  });
});

describe('newCardPage', () => {
  const props = {
    addPostAction: jest.fn(),
    isLoading: false,
  };

  const newCardPage = shallow(<NewCardPage {...props} />);

  it('snapshot', () => {
    expect(toJson(newCardPage)).toMatchSnapshot();
  });
});
