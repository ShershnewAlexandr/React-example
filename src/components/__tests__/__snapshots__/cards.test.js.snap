// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addCard snapshot 1`] = `
<div
  className="add-card-container"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="card-space"
    container={true}
    justify="center"
  >
    <Link
      className="link"
      to="/newcard"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <div
          className="card-plus-container"
        >
          <WithStyles(ForwardRef(Grid))
            alignItems="center"
            container={true}
            justify="center"
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <span
                className="card-plus"
              >
                +
              </span>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </div>
      </WithStyles(ForwardRef(Grid))>
    </Link>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`card snapshots 1`] = `
<div
  className="card-container"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-start"
    container={true}
    direction="column"
    justify="space-between"
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      className="card-title-container"
      item={true}
    >
      <span
        className="card-title-text"
      >
        title
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="card-content-container"
      item={true}
    >
      <span
        className="card-content-text"
      >
        content text
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="card-buttons-container"
      container={true}
      direction="row"
      item={true}
      justify="flex-end"
    >
      <WithStyles(ForwardRef(Grid))
        data-testid="like-card-button"
        item={true}
      >
        <LikeButton
          id={5}
          likePostAction={[MockFunction]}
          liked={false}
          totalLikes={89}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`card snapshots 2`] = `
<div
  className="card-container"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-start"
    container={true}
    direction="column"
    justify="space-between"
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      className="card-title-container"
      item={true}
    >
      <span
        className="card-title-text"
      >
        title
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="card-content-container"
      item={true}
    >
      <span
        className="card-content-text"
      >
        content text
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="card-buttons-container"
      container={true}
      direction="row"
      item={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        data-testid="edit-card-button"
        item={true}
      >
        <EditButton
          id={5}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        data-testid="like-card-button"
        item={true}
      >
        <LikeButton
          id={5}
          likePostAction={[MockFunction]}
          liked={false}
          totalLikes={89}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`card snapshots 3`] = `
<div
  className="card-container"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-start"
    container={true}
    direction="column"
    justify="space-between"
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      className="card-title-container"
      item={true}
    >
      <span
        className="card-title-text"
      >
        title
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="card-content-container"
      item={true}
    >
      <span
        className="card-content-text"
      >
        content text
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="card-buttons-container"
      container={true}
      direction="row"
      item={true}
      justify="flex-end"
    >
      <WithStyles(ForwardRef(Grid))
        data-testid="like-card-button"
        item={true}
      >
        <LikeButton
          id={5}
          likePostAction={[MockFunction]}
          liked={true}
          totalLikes={89}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`cardForm snapshot 1`] = `undefined`;

exports[`cardListPage snapshots 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    item={true}
    justify="flex-end"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={1}
    >
      <Link
        to="/signin"
      >
        <span>
          cardList.signOut
        </span>
      </Link>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-start"
    className="cardlist-container"
    container={true}
    direction="row"
    item={true}
    justify="space-evenly"
  />
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Connect(CardPageBarContainer) />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`cardListPage snapshots 2`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    item={true}
    justify="flex-end"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={1}
    >
      <Link
        to="/signin"
      >
        <span>
          cardList.signOut
        </span>
      </Link>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-start"
    className="cardlist-container"
    container={true}
    direction="row"
    item={true}
    justify="space-evenly"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <AddCard />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Connect(CardPageBarContainer) />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`editButton snapshot 1`] = `
<Link
  onClick={[Function]}
  to="/editcard/5"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="edit-button-container"
    container={true}
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <i
        className="material-icons md-24"
      >
        edit
      </i>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Link>
`;

exports[`editCardPage snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="editCardWrapper-container"
    container={true}
    direction="column"
    item={true}
    justify="space-around"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <span
        className="editCardWrapper-title"
      >
        editCard.cardInfo
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Connect(ReduxForm) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      spacing={2}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <button
          className="editCardWrapper-saveButton"
          onClick={[Function]}
          type="submit"
        >
          editCard.save
        </button>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <button
          className="editCardWrapper-deleteButton"
          onClick={[Function]}
          type="submit"
        >
          editCard.delete
        </button>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Link
          to="/cardlist"
        >
          <button
            className="editCardWrapper-CancelButton"
            type="submit"
          >
            editCard.cancel
          </button>
        </Link>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`likeButton snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="like-button-container"
  container={true}
  data-testid="likeButton"
  direction="row"
  justify="space-around"
  onClick={[Function]}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <span>
      23
    </span>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <i
      className="material-icons md-24"
      data-testid="like-icon"
    >
      thumb_up
    </i>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`newCardPage snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="newCardWrapper-container"
    container={true}
    direction="column"
    item={true}
    justify="space-around"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <span
        className="newCardWrapper-title"
      >
        newCard.newCard
      </span>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <ReduxForm
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="addPost"
        getFormState={[Function]}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        validate={[Function]}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      spacing={2}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <button
          className="newCardWrapper-addButton"
          onClick={[MockFunction]}
          type="submit"
        >
          newCard.add
        </button>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Link
          to="/cardlist"
        >
          <button
            className="editCardWrapper-CancelButton"
            type="submit"
          >
            newCard.cancel
          </button>
        </Link>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
